src/main/java/com/agency04/sbss/pizza/rest/CustomerRestController.java

```diff
@@ -35,30 +32,32 @@ public Customer getCustomer(@PathVariable String username) {
 
     @DeleteMapping("/{username}")
     public ResponseEntity deleteCustomer(@PathVariable String username) {
-        Customer customer = customers.stream().filter(x -> x.getUserName().equals(username)).findFirst().orElse(null);
-        if(customer == null){
+        Customer tempCustomer = customerRepository.findByUsername(username);
+
+        if (tempCustomer == null) {
             throw new CustomerNotFoundException("Customer with username " + username + " doesn't exist.");
         }
-        customers.remove(customer);
+
+        customerRepository.deleteByUsername(username);
         return ResponseEntity.ok(HttpStatus.OK);
     }
 
     @PostMapping("/")
     public ResponseEntity registerCustomer(@RequestBody Customer customer) {
-        customers.add(new Customer(customer));
-        if(customers.stream().filter(x -> x.getCustomerId() == customer.getCustomerId()).findFirst().orElse(null) == null ){
+        customer.setId(0);
+        customerRepository.save(customer);
+        if(customerRepository.findByUsername(customer.getUsername()) == null ){
             throw new CustomerNotFoundException("Customer is not registered.");
         }
         return  ResponseEntity.ok(HttpStatus.CREATED).status(201).build();
     }
 
     @PutMapping("/")
     public ResponseEntity updateCustomer(@RequestBody Customer customer) {
-        Customer temp = customers.stream().filter(x -> x.getCustomerId() == customer.getCustomerId()).findFirst().orElse(null);
-        if (temp != null) {
-            int index = customers.indexOf(temp);
-            customers.set(index, customer);
-            if(Objects.equals(customers.get(index), customer)){
+        Optional<Customer> tempCustomer = customerRepository.findById(customer.getId());;
+        if (tempCustomer.isPresent()) {
+            customerRepository.save(tempCustomer.get());
+            if(customer.equals(customerRepository.findById(customer.getId()))){
```

Comment:
Prvo spremiš korisnika i nakon toga ideš u bazu provjeriti postoji li takav korisnik s tim ID-em i zatim s equals provjeravaš jesu li objetki jednaki. Ima li to smisla? Dodati form objekt koji neće sadržavati id property, te zatim pretraživati po usernameu.